frontend/catalyst/compiler.py

```diff
@@ -484,9 +487,6 @@ def run_from_ir(self, ir: str, module_name: str, workspace: Directory):
             output = LinkerDriver.run(output_object_name, options=self.options)
             output_object_name = str(pathlib.Path(output).absolute())
 
-        # Clean up temporary files
-        if os.path.exists(tmp_infile_name):
-            os.remove(tmp_infile_name)
```

Comment:
Why is this no longer needed?

Reply:
I changed it back just before you mentioned. I ideally want to remove this if possible since I don't want to spend time removing files that will be removed entirely by the temporary directory. Yes they might not be nice to have when the user has `keep_intermediate=True`, but I am still thinking of the best way to go around that.